server/src/main/java/com/group10/lobcitydata/services/RapidApiAdaptor.java

```diff
@@ -1,85 +1,87 @@
 import com.group10.lobcitydata.models.rapidapi.Team;
-import org.springframework.asm.TypeReference;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
 
+import java.io.IOException;
 import java.net.URI;
 import java.net.http.HttpClient;
 import java.net.http.HttpRequest;
 import java.net.http.HttpResponse;
 import java.util.List;
-import java.util.stream.Collectors;
+import java.util.Map;
 
 @Service
 public class RapidApiAdaptor {
     private final RapidApiConfig config;
 
+    private static final String TEAMS_PATH = "/teams";
+    private static final String PLAYERS_PATH = "/players";
+
     @Autowired
     public RapidApiAdaptor(RapidApiConfig config) {
         this.config = config;
     }
 
-    public List<Team> getTeams() throws Exception {
-        // Build URL for the request, I will probaby build something to make this nicer.
-        StringBuilder reqBuilder = new StringBuilder();
-        reqBuilder.append(config.getUrlBase());
-        reqBuilder.append("/teams");
-
-        // Build the request to the external API
-        HttpRequest request = HttpRequest.newBuilder()
-                .uri(URI.create(reqBuilder.toString()))
-                .header("X-RapidAPI-Key", config.getApiKey())
-                .header("X-RapidAPI-Host", config.getHost())
-                .method("GET", HttpRequest.BodyPublishers.noBody())
-                .build();
+    public List<Team> getTeams(Map<String, String> queryParams) throws Exception {
+        UrlBuilder ub = new UrlBuilder(config.getUrlBase(), TEAMS_PATH);
+        if (!queryParams.isEmpty()) {
+            ub.addQueryParam(queryParams);
+        }
 
-        // Execute the request and if we receive back a non 200 status code we will throw an error
-        var response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString());
+        HttpResponse<String> response = makeGetRequest(ub);
         if (!HttpStatus.valueOf(response.statusCode()).is2xxSuccessful()) {
             throw new Exception("Received a bad status code, Response: " + response.body());
         }
 
-        // Map the JSON string data to Java classes
-        ObjectMapper mapper = new ObjectMapper();
-        JavaType type = mapper.getTypeFactory().
-                constructParametricType(ApiResponse.class, Team.class);
+        return mapResponse(Team.class, response.body()).getResponse();
+    }
 
-        ApiResponse<Team> formattedResponse = new ObjectMapper()
-                .readerFor(type)
-                .readValue(response.body());
+    public List<Player> getPlayers(Map<String, String> queryParams) throws Exception {
+        UrlBuilder ub = new UrlBuilder(config.getUrlBase(), PLAYERS_PATH);
+        ub.addQueryParam("country", "USA");
+        if (!queryParams.isEmpty()) {
+            ub.addQueryParam(queryParams);
+        }
```

Comment:
We might need to tweak this to get access to non-USA players with specific queries. Not entirely sure though.
```suggestion
        UrlBuilder ub = new UrlBuilder(config.getUrlBase(), PLAYERS_PATH);
        if (!queryParams.isEmpty()) {
            ub.addQueryParam(queryParams);
        } else ub.addQueryParam("country", "USA");
```